Ignore unsafe rules from check --fix

lsegal opened the pull request:
Seed with some rules we do not "trust" to work reliably. Respects `--unsafe` flag.

Also fixes an eslint issue in converting char ranges to bytes for replacements.

brynary reviewed (approved):
PR looks good.

One quick clarification... Am I reading this PR and the code correctly that prior to this patch the `--unsafe` option did not have any impact on _non-LLM_ fixes?

(I think we added the concept of "unsafe" when we introduced LLM fixes, and before we thought about safe vs. unsafe _tool-generated_ fixes, so that would make sense given the evolution.)

lsegal commented:
> (I think we added the concept of "unsafe" when we introduced LLM fixes, and before we thought about safe vs. unsafe tool-generated fixes, so that would make sense given the evolution.)

Good catch I didn't realize this was already wired up through check. Updated with unsafe support.
